### ISSUE TITLE
Unset environment variable for OpenM++ in AAW

### ISSUE THREAD

**Issue by rochellegarner** [StatCan/aaw-kubeflow-containers#353]
To have multiple versions of OpenM++ on our virtual machines in AAW, we need to be able to set the OM_ROOT environment variable. At the global level it's set to /opt/openmpp and that's something namespace users can't change. We would need to unset this so that things would work the way the OpenM++ developers had intended when writing some scripts to open the UI. @goatsweater or @YannCoderre , can either of you help with this?

**Comment by goatsweater** [comment]
I haven't been able to make any progress on this. The repo assumes a hard dependency on `docker`, but I've only got `nerdctl` installed. Tried to alias docker to nerctl since they provide pretty much the same command set, but aliases aren't expanded in Makefiles:

```
aaw-kubeflow-containers git:(master) which docker
docker: aliased to nerdctl

aaw-kubeflow-containers git:(master) make build/remote-desktop
# End repo with exactly one trailing slash, unless it is empty
REPO=$(echo "k8scc01covidacr.azurecr.io" | sed 's:/*$:/:' | sed 's:^\s*/*\s*$::') &&\
	IMAGE_NAME="${REPO}remote-desktop:master" && \
	docker build  --rm --force-rm -t $IMAGE_NAME ./output/remote-desktop && \
	echo -n "Built image $IMAGE_NAME of size: " && \
	docker images $IMAGE_NAME --format "{{.Size}}" && \
	echo "::set-output name=full_image_name::$IMAGE_NAME" && \
	echo "::set-output name=image_tag::master" && \
	echo "::set-output name=image_repo::${REPO}"
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
make: *** [build/remote-desktop] Error 1
```

**Comment by Jose-Matsuda** [comment]
@goatsweater if you want to and are just testing locally, you don't _haaave_ to use the makefile.
to be completely honest I don't use the makefile aside from doing `make generate-dockerfiles` and personally just do 

`docker build -t test output/remote-desktop/` and then do whatever Blair put in the makefile for the `make dev`. So I'd imagine you can do the same with nerdctl.

so I guess something like

`nerdctl build -t test output/remote-desktop/` and then to run
`nerdctl run -it --rm -p 8888:8888 -e NB_PREFIX=/notebook/username/notebookname test` and then navigate to `http://localhost:8888/notebook/username/notebookname` or something like that

**Comment by goatsweater** [comment]
Looks like a no go on M1 based images. There's some base images that aren't `aarch64` compatible so it builds an `amd64` image, unfortunately.

```
REPOSITORY              TAG       IMAGE ID        CREATED          PLATFORM       SIZE         BLOB SIZE
test                    latest    4ef3b9404a77    2 minutes ago    linux/amd64    12.4 GiB     3.9 GiB
```

Running looks like it works - I can get to a noVNC screen on `localhost:8888`. I get a "Failed to connect to server" message when I click connect.

**Comment by blairdrummond** [comment]
Are you using the full url path? Like
 http://localhost:8888/notebook/username/notebookname

Also the Makefile is due for an overhaul. I wrote the first version in Make but it was extended a lot afterwards to include more testing and automation --- that's good, but Taskfile or something might be more appropriate. I also don't think we ever planned for multi-arch support 😛

**Comment by goatsweater** [comment]
You were right - missed the full path. I am able to get into the system, and starting OpenM++ from the icon did work. I don't think that really means multi-arch really works, but at least while rosetta 2 is around you can run an amd64 version of an AAW container through rancher desktop via `nerdctl`. 😀

**Comment by goatsweater** [comment]
I've got a working local test of OpenM++ where I've *unset* `$OM_ROOT` and things seem to be functioning as expected. Still to test is downloading and running a diff version to verify that my success isn't just a fluke.

**Comment by goatsweater** [comment]
@Souheil-Yazji As per the discussion this morning I've initiated the merge as #359.

**Comment by Souheil-Yazji** [comment]
@goatsweater @rochellegarner 

Issue should be resolved by PR #363. A new notebook should be created and used (image name = remote-desktop:v1). 
`$OM_ROOT` is unset on the new notebooks. M++ still appears to work.

For other versions of OMM (per @goatsweater comment):
- Download a different version of OpenM++
- Put a version of OpenM++ into a sub directory of /opt/openmpp/<version>
- Define a second desktop icon and point it to the new version then run it
